code/modules/projectiles/guns/energy.dm

```diff
@@ -29,69 +30,150 @@
 	else
 		power_supply = new(src)
 	power_supply.give(power_supply.maxcharge)
-	if(self_recharge)
+	var/obj/item/ammo_casing/energy/shot
+	for (var/i = 1, i <= ammo_type.len, i++)
+		var/shottype = ammo_type[i]
+		shot = new shottype(src)
+		ammo_type[i] = shot
+	shot = ammo_type[select]
+	fire_sound = shot.fire_sound
+	fire_delay = shot.delay
+	if(selfcharge)
 		processing_objects.Add(src)
 	update_icon()
 
 /obj/item/weapon/gun/energy/Destroy()
-	if(self_recharge)
+	if(selfcharge)
 		processing_objects.Remove(src)
 	return ..()
 
 /obj/item/weapon/gun/energy/process()
-	if(self_recharge) //Every [recharge_time] ticks, recharge a shot for the cyborg
+	if(selfcharge) //Every [recharge_time] ticks, recharge a shot for the cyborg
 		charge_tick++
-		if(charge_tick < recharge_time) return 0
+		if(charge_tick < charge_delay)
+			return 0
 		charge_tick = 0
-
-		if(!power_supply || power_supply.charge >= power_supply.maxcharge)
+		if(!power_supply)
 			return 0 // check if we actually need to recharge
-
+		var/obj/item/ammo_casing/energy/E = ammo_type[select]
 		if(use_external_power)
 			var/obj/item/weapon/stock_parts/cell/external = get_external_power_supply()
-			if(!external || !external.use(charge_cost/10)) //Take power from the borg...
+			if(!external || !external.use((E.e_cost)/10)) //Take power from the borg...
 				return 0								//Note, uses /10 because of shitty mods to the cell system
+		power_supply.give(E.e_cost) //... to recharge the shot

```

Comment:
Changing it to 1000, guns shouldn't charge faster based on the mode they're set to.